proposals/file-structure.md

```diff
@@ -0,0 +1,99 @@
+# File structure 
+## Issues with current file structure
+
+- Not consistent structure
+  - Some components is in app-packages, some are still in app
+  - Not clear rules for nesting levels
+- Co-location / Navigating
+  - Hard to understand dependencies, ex: what pages will be impacted by changing given given reducer, action or saga
+- Disposability
+  - Very hard to get rid / refactor parts of the system, as dependencies not clear
+- Code splitting
+  - Impossible to code split regarding not clear boundaries of files, needed for a given page
+- QA and AB testing
+  - Easy to leak logic to components level that are used in different pages
+  - **Hard for QA to identify the impact to business logic**
+- Dependencies
+  - High coupling between who uses given reducer, action, saga file
+## Feature drive architecture
+- A *Feature* is a self-contained user facing reusable complex building block
+- Provides out of the box 
+  - Disposability, modifiability - can easily remove modify a feature and not break unrelated logic
+  - code splitting - there is a logical separation to split a code in webpack on.
+  - co-location - all feature related code is in one place, it can be removed, other things will not crash
+  - Easy for QA to identify the business logic impact
+
+```
+ui-components
+packages
+	app-components
+app
+	shared
+		# Reusable functionality between features we don't want to put to app-components or ui-components
+		# In case for some reason we don't store it in ui-componenst and app-components
+		actions
+		    # redux.entities related actions, sagas and reducers
+		sagas
+		reducers
+	features
+		# A reusable complex user facing component with actions / reducers / actions included
+		# Feature cannot nest or depend on other feature
+		# If some redux branch is shared between different components, maybe they all belong to the same feature
+		# Idealy feature should have a isolated redux branch
+		Basket
+			actions
+			    # Actions, reducers and sagas that related to redux non entity branch	
+			reducers
+			sagas
+			components
+			    BasketTable
+			    ... 
+			containers
+			index.js
+    			# index.jsx is a contract of feature. 
+                # Can have multiple exports. (variations for iframe/hosted etc) 
+		Header
```

Comment:
`Basket` and `Header` are `app-components` concerns in my mind.

Perhaps as a rule `Header` should be in `app-components` as a set of components, then in `festicket-app` we have the data provider. 

Not sure how it would work for nested containers though.